src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -2872,26 +2880,8 @@ namespace InternalHeatGains {
                         }
 
                         // Environmental class
-                        if (UtilityRoutines::SameString(AlphaName(10), "None")) {
-                            thisZoneITEq.Class = ITEClassNone;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A1")) {
-                            thisZoneITEq.Class = ITEClassA1;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A2")) {
-                            thisZoneITEq.Class = ITEClassA2;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A3")) {
-                            thisZoneITEq.Class = ITEClassA3;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A4")) {
-                            thisZoneITEq.Class = ITEClassA4;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "B")) {
-                            thisZoneITEq.Class = ITEClassB;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "C")) {
-                            thisZoneITEq.Class = ITEClassC;
-                        } else {
-                            ShowSevereError(state, std::string{RoutineName} + itEqModuleObject + ": " + AlphaName(1));
-                            ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(10) + '=' + AlphaName(10));
-                            ShowContinueError(state, "Valid entries are None, A1, A2, A3, A4, B or C.");
-                            ErrorsFound = true;
-                        }
+                        thisZoneITEq.Class = getEnumerationValue(ITEClassNamesUC, UtilityRoutines::MakeUPPERCase(AlphaName(10)));
+                        ErrorsFound |= (thisZoneITEq.Class == static_cast<int>(ITEClass::Invalid));
```

Comment:
This should be `II=` which is boolean or. `|=` is bitwise or. Yes, they are the same in this specific case. But it's better to be precise.

Reply:
Actually here I took an advise from a discussion here here which is trying to say that `|=` is considered to be safer to use than `||=`--but I would say I did not fully understand why:
https://stackoverflow.com/questions/2488406/why-doesnt-c-have-or-for-booleans

Reply:
Actually, it seems that compiler (Windows MSVC) does not recognize `||=`:
![image](https://user-images.githubusercontent.com/64040111/179630910-ce4111a4-de58-4c17-b22d-4bda0f202aa4.png)

I have to use `|=` instead to pass compiling. Wonder if you guys got similar problems?

Reply:
TIL Weird!! In that case, I would do ErrorsFound = ErrorsFound || … Bitwise and Boolean operations should be the same for type bool, but for people who have done bit operations they look strange.